cli/cmd/migrate_webhooks.go

```diff
@@ -0,0 +1,464 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/keptn/go-utils/pkg/api/models"
+	api "github.com/keptn/go-utils/pkg/api/utils"
+	"github.com/keptn/keptn/cli/internal"
+	"github.com/keptn/keptn/cli/pkg/common"
+	"github.com/keptn/keptn/cli/pkg/credentialmanager"
+	"github.com/keptn/keptn/webhook-service/lib"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+type migrateWebhooksCmdParams struct {
+	ProjectName *string
+	DryRun      *bool
+}
+
+var migrateWebhooksParams *migrateWebhooksCmdParams
+var ErrResourceNotFound = fmt.Errorf("Resource not found")
+var supportedCurlMethods = [4]string{"POST", "PUT", "GET", "HEAD"}
+var webhookURI = "%2Fwebhook%2Fwebhook.yaml"
+
+const betaApiVersion = "webhookconfig.keptn.sh/v1beta1"
+const alphaApiVersion = "webhookconfig.keptn.sh/v1alpha1"
+
+type webhookResource struct {
+	Project         *string
+	Stage           *string
+	Sevice          *string
+	WebhookResource *models.Resource
+}
+
+var migrateWebhooksCmd = &cobra.Command{
+	Use:     "migrate-webhooks",
+	Short:   `Migrates the webhook configurations from version v1alpha1 to v1beta1`,
+	Example: `keptn migrate-webhooks [--project=PROJECTMNAME] [--dry-run]`,
+	Long: `Migrates the webhook configurations from version v1alpha1 to v1beta1. Version v1beta1 is the new default version
+				and it is highly encouraged to use this one. Version v1alpha1 won't be supported from version Keptn 0.19.0`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return doMigration(migrateWebhooksParams)
+	},
+}
+
+func doMigration(params *migrateWebhooksCmdParams) error {
+	endPoint, apiToken, err := credentialmanager.NewCredentialManager(assumeYes).GetCreds(namespace)
+	if err != nil {
+		return fmt.Errorf(authErrorMsg)
+	}
+
+	api, err := internal.APIProvider(endPoint.String(), apiToken)
+	if err != nil {
+		return internal.OnAPIError(err)
+	}
+
+	if *params.ProjectName != "" { // project for migration selected
+		project, err := checkProjectExists(*params.ProjectName, api)
+		if err != nil {
+			return fmt.Errorf("Failed to migrate webhook configurations for project %s: %s", *params.ProjectName, err.Error())
+		}
+
+		webhooks, err := getProjectWebhooks(project, api)
+		if err != nil {
+			return fmt.Errorf("cannot retrieve webhook-config for project %s: %s", *params.ProjectName, err.Error())
+		}
+		if err := migrateWebhooks(webhooks, params, api); err != nil {
+			return fmt.Errorf("cannot migrate webhook-config for project %s: %s", *params.ProjectName, err.Error())
+		}
+	} else { // project for migration not selected, migrating all projects
```

Comment:
shouldn't this behaviour be explained in the helper?

Reply:
Yes, I can add it there